### ISSUE TITLE
Setting value to variables using PyFMI

### ISSUE THREAD

**Issue by lymereJ** [lbl-srg/EnergyPlusToFMU#34]
I built an FMU of the `_fmu-export-variable.idf` example file and I am using PyFMI to run the simulation. I've noticed that the value of `yShadeFMU` is always 0. I've tried to modify it using `.set()` (see [here](https://jmodelica.org/pyfmi/pyfmi.html#pyfmi.fmi.ModelBase.set)) but `yShadeFMU` stays at 0. I've tried doing the same using a FMU shipped with PyFMI, and there it worked. I'm concerned that there might something wrong with my FMU. I've tried on different operating system and get the same result. Am I missing something?

**Comment by tsnouidui** [comment]
Which version of E+ are you using? Could you try the same example using the `_fmu-export-actautor.idf` instead. Recent versions of E+ are not supporting erl-variables with the EnergyPlusFMU well.

**Comment by tsnouidui** [comment]
@lymereJ any feedback on this? Otherwise, I will close the issue.

**Comment by lymereJ** [comment]
@tsnouidui, I have not tried your suggestion yet. I'll try this week and get back to you.

**Comment by lymereJ** [comment]
@tsnouidui, I'm using EnergyPlus 9.0 but I've also tried using version 8.9. I got the same results when trying the same example using the `_fmu_export_actuator.idf`.

**Comment by Denzo77** [comment]
@lymereJ, I had to pass yShade into the `input` parameter of `model.simulate()` to get it to work. I've uploaded an example [here](https://gist.github.com/Denzo77/ce5d15509be6017884d981e31a4cd2eb).

**Comment by lymereJ** [comment]
@Denzo77, thanks for getting back to me! I actually did try using `input` as an argument in `model.simulate()` using a slightly different approach than what you proposed in your gist (see below) and ... It was actually working :roll_eyes:. I was under the impression that it wasn't working because `res['yShade']` is always return an array of 0s, I got so fixated on it that I didn't notice that it actually did have an impact on `res['TRoo']`. Anyway, thanks for bringing this topic back up and providing an answer!

For what it's worth, here's the approach I was following using the `input` argument:
```
def f(time):
    return 6

input_object = ('yShade', f)

res = model.simulate(start_time=0, final_time=60*60*24*3, input=input_object, options=opts)
```
@tsnouidui is it expected behavior to return an array of 0s?

**Comment by JChung-rpi** [comment]
Hi @lymereJ , I am wondering if you have solved this issue. I have been struggling with the same one for the past week. I have attached the figure I have generated using @Denzo77 's codes. I am using EnergyPlus v.9.4 which is the same version as the '_fmu-export-actuator.idf' file.

![screenshot](https://github.com/lbl-srg/EnergyPlusToFMU/assets/93723525/b3f772d9-273e-4a20-bc1c-f94cf7bfb819)

**Comment by lymereJ** [comment]
@JChung-rpi - Wow, that's a blast from the past!

I think that I did find a solution but I don't have that particular example anymore, I think., I've moved on to using the EnergyPlus Python API for newer projects. That being said, in a script that I still have on my machine (which is a bit more complex than the example) and also relies on PyFMI, it looks like I ended up using `do_step()` instead of `simulate()`, I think that was part of the solution since I see `set()` being used in the script.
